tests/MongoDB.Driver.Tests/Specifications/crud/AggregateTest.cs

```diff
@@ -59,16 +62,31 @@ protected override List<BsonDocument> ConvertExpectedResult(BsonValue expectedRe
             return ((BsonArray)expectedResult).Select(x => (BsonDocument)x).ToList();
         }
 
-        protected override List<BsonDocument> ExecuteAndGetResult(IMongoCollection<BsonDocument> collection, bool async)
+        protected override List<BsonDocument> ExecuteAndGetResult(IMongoDatabase database, IMongoCollection<BsonDocument> collection, bool async)
         {
-            if (async)
+            if (collection == null)
```

Comment:
This same class is used to invoke database.Aggregate (when collection is null) or collection.Aggregate (when collection is specified).